src/openepd/api/average_dataset/generic_estimate_sync_api.py

```diff
@@ -0,0 +1,206 @@
+#
+#  Copyright 2024 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+from typing import Literal, TypeAlias, overload
+
+from requests import Response
+
+from openepd.api.base_sync_client import BaseApiMethodGroup
+from openepd.api.common import StreamingListResponse
+from openepd.api.dto.common import BaseMeta, OpenEpdApiResponse
+from openepd.api.dto.meta import PagingMeta, PagingMetaMixin
+from openepd.api.utils import encode_path_param
+from openepd.model.generic_estimate import (
+    GenericEstimate,
+    GenericEstimatePreview,
+    GenericEstimateRef,
+    GenericEstimateWithDeps,
+)
+
+
+class GenericEstimateApi(BaseApiMethodGroup):
+    """API methods for Generic Estimates."""
+
+    @overload
+    def get_by_open_xpd_uuid(self, uuid: str, with_response: Literal[True]) -> tuple[GenericEstimate, Response]: ...
+
+    @overload
+    def get_by_open_xpd_uuid(self, uuid: str, with_response: Literal[False] = False) -> GenericEstimate: ...
+
+    def get_by_open_xpd_uuid(
+        self, uuid: str, with_response: bool = False
+    ) -> GenericEstimate | tuple[GenericEstimate, Response]:
+        """
+        Get Generic Estimates by OpenEPD UUID.
+
+        :param uuid: OpenEPD UUID
+        :return: GE
+        :raise ObjectNotFound: if Generic Estimates is not found
+        """
+        response = self._client.do_request("get", f"/generic_estimates/{uuid}")
```

Comment:
You aren't converting `HTTPError` into `ObjectNotFound`.

Reply:
This happens inside of client implementation: 

```
def _do_request...
        error_handler = self._http_error_handlers.get(response.status_code, None)
        if error_handler:
            response = error_handler(response, raise_for_status)

...

@staticmethod
def handle_not_found(response: Response, raise_for_status: bool) -> Response | None:

```